components/clp-py-utils/clp_py_utils/clp_config.py

```diff
@@ -355,6 +359,27 @@ def validate_logs_dir(self):
         except ValueError as ex:
             raise ValueError(f"logs_directory is invalid: {ex}")
 
+    def load_execution_container_name(self):
+        if self.execution_container is not None:
+            # Accept configured value in clp-config.yml for debug purposes
+            return
+
+        with open(self.os_release_file_path) as os_release_file:
+            parsed = {}
+            for line in os_release_file:
+                var, val = line.strip().split("=")
+                parsed[var] = val
+
+            self.execution_container = "ghcr.io/y-scope/clp/"
+            if "ubuntu" == parsed["ID"]:
```

Comment:
can we store parsed["ID"] in a local variable?

Reply:
When I drafted the code, I thought about extracting those into variables `parsed_id`, `parsed_version_id` / `parsed_version_codename`, and even better we `break` the for-loop if all info we need has been parsed. However, that might be less clean and less extensible if we (ever) plan to support CentOS / RHEL in the future. That's because `VERSION_ID` is common across all Linux distributions, but `VERSION_CODENAME` is specific to Ubuntu / Debian distributions.